fem/fe.cpp

```diff
@@ -722,17 +722,34 @@ void NodalFiniteElement::Project(
 {
    if (fe.GetRangeType() == SCALAR)
    {
-      MFEM_ASSERT(map_type == fe.GetMapType(), "");
```

Comment:
In the non-scalar (vector) case, what happens if `map_type != fe.GetMapType()`? Is that case supported, or does there need to be an assert?

Reply:
In the vector case `fe.GetMapType()` will either be H_CURL or H_DIV whereas `map_type` will be either VALUE or INTEGRAL.  So, these are guaranteed to be different.  We handle both H_CURL and H_DIV by calling `fe.CalcVShape` with the `ElementTransformation` argument which returns shape functions transformed to the physical space where the map type distinction doesn't matter.  The VALUE or INTEGRAL type for the nodal finite element basis functions is handled by scaling with the Jacobian determinant on line 767.  In short, all possibilities are handled but I grant you that this is far from obvious.